app/controllers/application_templates_controller.rb

```diff
@@ -27,7 +27,7 @@ def toggle_active
     else show_message :inactive_application,
                       default: 'The application is now inactive.'
     end
-    redirect_to :back
+    redirect_back fallback_location: main_dashboard_path
```

Comment:
You know, actually, you're right. Seems like we shouldn't have the same fallback location everywhere. Here, for example, the main dashboard isn't where we would expect `:back` to normally be.

In general, shouldn't the fallback be wherever we expect they've come from?